core/templates/dev/head/pages/admin/misc_tab/AdminMiscTabDirective.js

```diff
@@ -76,6 +79,54 @@ oppia.directive('adminMiscTab', [
         $scope.downloadTopicSimilaritiesFile = function() {
           $window.location.href = ADMIN_TOPICS_CSV_DOWNLOAD_HANDLER_URL;
         };
+
+        var setDataExtractionQueryStatusMessage = function(message) {
+          $scope.showDataExtractionQueryStatus = true;
+          $scope.dataExtractionQueryStatusMessage = message;
+        };
+
+        $scope.submitQuery = function() {
+          var STATUS_PENDING = (
+            'Data extraction query has been submitted. Please wait.');
+          var STATUS_FINISHED = 'Loading the extracted data ...';
+          var STATUS_FAILED = 'Error, ';
+
+          var data = {
+            exp_id: $scope.expId,
+            exp_version: $scope.expVersion,
+            state_name: $scope.stateName,
+            num_answers: $scope.numAnswers
+          };
+
+          setDataExtractionQueryStatusMessage(STATUS_PENDING);
+
+          $http.post(
+            DATA_EXTRACTION_QUERY_HANDLER_URL, data
+          ).then(
+            function(response) {
+              var data = response.data;
+              setDataExtractionQueryStatusMessage(STATUS_FINISHED);
+              var extractedAnswers = JSON.stringify(data.data);
+              var dataURL = (
+                'data:application/json;base64,' + btoa(extractedAnswers));
+              var downloadURL = (
+                '<a href="' + dataURL +
```

Comment:
Oh gosh, I can't believe I missed this. No, please do not construct HTML in the JS. You're gonna run into security issues that way.

Perhaps take a look at how it's done in the exploration history tab?